llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7722,6 +7722,7 @@ DenseMap<const SCEV *, Value *> LoopVectorizationPlanner::executePlan(
   BestVPlan.prepareToExecute(ILV.getTripCount(),
                              ILV.getOrCreateVectorTripCount(nullptr),
                              CanonicalIVStartValue, State);
+  VPlanTransforms::prepareToExecute(BestVPlan);
```

Comment:
Better rename this VPlan2VPlan conversion pass, say, `convertToConcreteRecipes()`, and place it above, following VPlanTransforms::optimizeForVFAndUF(), rather than here, in the midst of `State`-dependent code generation? (BestVPlan.prepareToExecute() actually starts to generate code.)

Reply:
Restored the original position for now, will need to check where there are remaining users of getCanonicalIV, and how to best update them